src/compiler/checker.ts

```diff
@@ -26339,6 +26344,16 @@ namespace ts {
                 });
             }
 
+            if (hasNonAmbientClass && !isConstructor && symbol.flags & SymbolFlags.Function) {
+                // A non-ambient class cannot be an implementation for a non-constructor function/class merge
+                // TODO: The below just replicates our older error from when classes and functions were

```

Comment:
I would drop the TODO from the code and either put a comment in this PR or create a bug.

Reply:
If you can come up with a decentish new error message, I can add it, if you'd like - otherwise I'll just open an issue for the error message with a `help wanted` tag.

Reply:
I vote for creating an issue since I need to spend the rest of my afternoon on reviewing PRs before the RC snap.

Reply:
#32795